torch/distributed/_pycute/layout.py

```diff
@@ -38,39 +35,61 @@
 """
 
 from itertools import chain
-
-from .int_tuple import *
+from typing import Optional, TypeGuard, Union
+
+from .int_tuple import (
+    crd2idx,
+    flatten,
+    has_none,
+    IntTuple,
+    is_int,
```

Comment:
Please explicitly type as TypeAlias. Removes ambiguity for type checker and IDEs.